crates/starknet-server/tests/test_dump_and_load.rs

```diff
@@ -0,0 +1,272 @@
+pub mod common;
+
+// Important! Use unique file names for dump files, tests can be run in parallel.
+mod dump_and_load_tests {
+    use std::fs::{self};
+    use std::path::Path;
+    use std::process::Command;
+
+    use hyper::{Body, StatusCode};
+    use serde_json::json;
+    use starknet_rs_providers::Provider;
+    use starknet_types::felt::Felt;
+
+    use crate::common::devnet::BackgroundDevnet;
+    use crate::common::utils::get_json_body;
+
+    static DUMMY_ADDRESS: &str = "0x1";
+    static DUMMY_AMOUNT: u128 = 1;
+
+    use std::sync::Arc;
+
+    use starknet_rs_accounts::{Account, ExecutionEncoding, SingleOwnerAccount};
+    use starknet_rs_contract::ContractFactory;
+    use starknet_rs_core::chain_id;
+    use starknet_rs_core::types::{BlockId, BlockTag, FieldElement};
+    use starknet_rs_signers::{LocalWallet, SigningKey};
+
+    use crate::common::utils::get_events_contract_in_sierra_and_compiled_class_hash;
+
+    #[tokio::test]
+    async fn mint_dump_on_transaction_and_load() {
+        // dump after transaction
+        let dump_file_name = "dump_on_transaction";
+        let devnet_dump = BackgroundDevnet::spawn(Some(
+            [
+                "--dump-path".to_string(),
+                dump_file_name.to_string(),
+                "--dump-on".to_string(),
+                "transaction".to_string(),
+            ]
+            .to_vec(),
+        ))
+        .await
+        .expect("Could not start Devnet");
+        let req_body = Body::from(
+            json!({
+                "address": DUMMY_ADDRESS,
+                "amount": DUMMY_AMOUNT
+            })
+            .to_string(),
+        );
+        let resp = devnet_dump.post_json("/mint".into(), req_body).await.unwrap();
+        assert_eq!(resp.status(), StatusCode::OK, "Checking status of {resp:?}");
+        let mut resp_body = get_json_body(resp).await;
+        let tx_hash_value = resp_body["tx_hash"].take();
```

Comment:
No need to use `.take()`, here nor in the other place where it's used. `take` leaves a `Null` in the place of the taken value. What I would recommend is to use `resp_body["tx_hash"].as_str().unwrap()`, because it will also allow to replace the horribly long `FieldElement::from_hex_be(tx_hash_value.as_str().unwrap()).unwrap()` with a simpler `FieldElement::from_hex_be(tx_hash_value).unwrap()`.

Reply:
fixed in f3dd993b86e2dfe0e1a3db4d65be92d8f9aa655d